src/graph/graphNodes.ts

```diff
@@ -0,0 +1,128 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as vscode from 'vscode';
+import * as path from 'path';
+import { Command } from 'vscode';
+import { DocumentClient } from 'documentdb';
+import { INode } from '../nodes';
+import * as util from "./../util";
+import { GraphViewsManager } from "./GraphViewsManager";
+import { GraphConfiguration } from './GraphConfiguration';
+import { GraphViewServer } from './GraphViewServer';
+
+export class GraphDatabaseNode implements INode {
+	public readonly contextValue: string = "cosmosGraphDatabase";
+
+	private _graphEndpoint: string;
+	private _graphPort: number;
+
+	constructor(readonly id: string, private readonly _masterKey: string, private readonly _documentEndpoint: string, private readonly _graphViewsManager: GraphViewsManager, readonly server: INode) {
+		this._parseEndpoint(_documentEndpoint);
+	}
+
+	private _parseEndpoint(documentEndpoint: string): void {
+		// Document endpoint: https://<graphname>.documents.azure.com:443/
+		// Gremlin endpoint: <graphname>.graphs.azure.com
+		let [, address, , port] = this._documentEndpoint.match(/^[^:]+:\/\/([^:]+)(:([0-9]+))?\/?$/);
+		this._graphEndpoint = address.replace(".documents.azure.com", ".graphs.azure.com");
+		console.assert(this._graphEndpoint.match(/\.graphs\.azure\.com$/), "Unexpected endpoint format");
+		this._graphPort = parseInt(port || "443");
+		console.assert(this._graphPort > 0, "Unexpected port");
+	}
+
+	get masterKey(): string {
+		return this._masterKey;
+	}
+
+	get documentEndpoint(): string {
+		return this._documentEndpoint;
+	}
+
+	get graphEndpoint(): string {
+		return this._graphEndpoint;
+	}
+
+	get graphPort(): number {
+		return this._graphPort;
+	}
+
+	get label(): string {
+		return this.id;
+	}
+
+	get graphViewsManager(): GraphViewsManager {
+		return this._graphViewsManager;
+	}
+
+	get iconPath(): any {
+		return {
+			light: path.join(__filename, '..', '..', '..', '..', 'resources', 'icons', 'theme-agnostic', 'Azure Graph - database LARGE.svg'),
```

Comment:
Were you intending to add 'Azure Graph - database LARGE' in this PR? I don't see it

Reply:
TODO